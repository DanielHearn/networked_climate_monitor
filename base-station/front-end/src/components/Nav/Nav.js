import vButton from './../vButton/vButton.vue'
import { mapActions } from 'vuex'

export default {
  name: 'v-nav',
  components: {
    vButton
  },
  props: {
    loggedIn: {
      type: Boolean,
      default: false,
      required: true
    },
    mobile: {
      type: Boolean,
      default: false,
      required: true
    },
    mobileMenu: {
      type: Boolean,
      default: false,
      required: true
    }
  },
  methods: {
    ...mapActions(['logout']),
    toggleMenu: function() {
      this.$store.commit('setMobileMenu', !this.$store.state.mobileMenu)
    }
  }
}
