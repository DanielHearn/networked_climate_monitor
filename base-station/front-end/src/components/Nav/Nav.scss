@import "./../../scss/variables.scss";

.nav {
  background: $accent-dark;
  color: $white;
  display: grid;
  grid-template-columns: repeat(3,1fr);
  padding: .25em .5em;
  flex: 1 auto;
  max-height: 3em;
  width: 100%;
  box-sizing: border-box;
  .nav__side {
    display: flex;
    flex-direction: row;
    padding: 0 0.2em;
    align-items: center;
    .title--nav {
      color: $white;
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
    p {
      margin: 0;
    }
  }
  .nav__left {
    justify-content: flex-start;
  }
  .nav__right {
    justify-content: flex-end;
  }
  .nav__links {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    display: flex;
    .link {
      padding: 0.5em 0.6em;
      justify-content: center;
      align-items: center;
      display: flex;
      color: $white;
      text-decoration: none;
      font-weight: $medium-weight;
      &.router-link-exact-active {
        text-decoration: underline;
      }
      &:hover {
        text-decoration: underline;
      }
    }
  }
}