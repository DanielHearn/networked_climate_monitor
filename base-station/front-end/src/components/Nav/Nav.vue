<template>
  <div class="nav">
    <div class="nav__side nav__left">
      <router-link to="/" class="link title--nav" v-if="!mobile"
        >Climate Monitor</router-link
      >
      <v-button
        v-else-if="loggedIn"
        @click.native="logout"
        class="logout-button"
        :hierachyLevel="'secondary'"
        :text="'person'"
        :isIcon="true"
      />
    </div>
    <div class="nav__links">
      <template v-if="!mobile">
        <router-link to="/" class="link">Home</router-link>
        <router-link to="/login" v-if="!loggedIn" class="link"
          >Login</router-link
        >
        <router-link to="/register" v-if="!loggedIn" class="link"
          >Register</router-link
        >
        <router-link to="/dashboard" v-if="loggedIn" class="link"
          >Dashboard</router-link
        >
        <router-link to="/settings" v-if="loggedIn" class="link"
          >Settings</router-link
        >
      </template>
    </div>
    <div class="nav__side nav__right">
      <v-button
        v-if="loggedIn && !mobile"
        @click.native="logout"
        class="logout-button"
        :hierachyLevel="'secondary'"
        :text="'person'"
        :isIcon="true"
        style="margin-left: 2em;"
      />
      <v-button
        v-if="mobile"
        class="mobile-menu-button"
        @click.native="toggleMenu"
        :hierachyLevel="'secondary'"
        :text="mobileMenu ? 'close' : 'menu'"
        :isIcon="true"
        style="margin-left: 2em;"
      />
    </div>
  </div>
</template>

<script src="./Nav.js"></script>

<style lang="scss" src="./Nav.scss"></style>
