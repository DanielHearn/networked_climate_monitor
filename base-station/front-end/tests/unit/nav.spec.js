import Vuex from 'vuex'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Nav from './../../src/components/Nav/Nav.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

const mutations = {
  setMobileMenu: jest.fn()
}

describe('Nav.vue', () => {
  let wrapper
  let store

  beforeEach(() => {
    store = new Vuex.Store({ mutations })
    wrapper = shallowMount(Nav, {
      propsData: {
        mobile: false,
        mobileMenu: false,
        loggedIn: false
      },
      stubs: ['router-link'],
      store,
      localVue
    })
  })

  afterEach(() => {
    wrapper.destroy()
    jest.resetAllMocks()
  })

  it('Desktop', async () => {
    expect(wrapper.find('.title--nav').exists()).toBe(true)
    expect(wrapper.find('.nav__links .link').exists()).toBe(true)
    expect(wrapper.find('.nav__links .link').text()).toBe('Home')
    expect(wrapper.find('.nav__right .logout-button').exists()).toBe(false)
    expect(wrapper.find('.nav__left .logout-button').exists()).toBe(false)
    expect(wrapper.find('.mobile-menu-button').exists()).toBe(false)

    wrapper.setProps({ loggedIn: true })
    expect(wrapper.find('.nav__right .logout-button').exists()).toBe(true)
    expect(wrapper.find('.nav__left .logout-button').exists()).toBe(false)
  })

  it('Mobile', async () => {
    wrapper.setProps({ mobile: true })
    expect(wrapper.find('.title--nav').exists()).toBe(false)
    expect(wrapper.find('.nav__links .link').exists()).toBe(false)

    expect(wrapper.find('.nav__right .logout-button').exists()).toBe(false)
    expect(wrapper.find('.nav__left .logout-button').exists()).toBe(false)

    expect(wrapper.find('.mobile-menu-button').exists()).toBe(true)

    wrapper.setProps({ loggedIn: true })
    expect(wrapper.find('.nav__right .logout-button').exists()).toBe(false)
    expect(wrapper.find('.nav__left .logout-button').exists()).toBe(true)
  })

  it('Mobile menu toggle', async () => {
    wrapper.setProps({ mobile: true })
    const menuButton = wrapper.find('.mobile-menu-button')
    expect(menuButton.exists()).toBe(true)
    expect(wrapper.find('.title--nav').exists()).toBe(false)
    menuButton.trigger('click')
    await wrapper.vm.$nextTick()
    expect(mutations.setMobileMenu).toHaveBeenCalledTimes(1)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
