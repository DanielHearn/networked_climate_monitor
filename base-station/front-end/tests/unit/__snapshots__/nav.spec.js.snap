// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm.vue has the expected html structure 1`] = `
<div
  class="nav"
>
  <div
    class="nav__side nav__left"
  >
    <router-link-stub
      class="link title--nav"
      to="/"
    >
      Climate Monitor
    </router-link-stub>
  </div>
   
  <div
    class="nav__links"
  >
    <router-link-stub
      class="link"
      to="/"
    >
      Home
    </router-link-stub>
     
    <router-link-stub
      class="link"
      to="/login"
    >
      Login
    </router-link-stub>
     
    <router-link-stub
      class="link"
      to="/register"
    >
      Register
    </router-link-stub>
     
    <!---->
     
    <!---->
  </div>
   
  <div
    class="nav__side nav__right"
  >
    <!---->
     
    <!---->
  </div>
</div>
`;
